### PR TITLE
Add support for POST request method in v1 api for entity detection

### PR DESCRIPTION
**Pull request by ankur09011**
## JIRA Ticket Number

JIRA TICKET: ML-2168

## Description of change
Add support for POST request method in v1 apis for entity detection.

## Checklist (OPTIONAL):

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Review by amansrivastava17 — APPROVED** [review]
Looks Good To Me

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Review by amansrivastava17 — DISMISSED** [review]

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v1/api.py [hunk 1]
```diff
@@ -290,8 +294,14 @@ def phone_number(request):
...
-        parameters_dict = get_parameters_dictionary(request)
-        ner_logger.debug('Start: %s ' % parameters_dict[PARAMETER_ENTITY_NAME])
+
+        parameters_dict = {}
+        if request.method == "POST":
+            parameters_dict = parse_post_request(request)
+            ner_logger.debug('Start POST %s ' % parameters_dict[PARAMETER_ENTITY_NAME])
+        elif request.method == "GET":
+            parameters_dict = get_parameters_dictionary(request)
+            ner_logger.debug('Start: GET %s ' % parameters_dict[PARAMETER_ENTITY_NAME])
```
Comment (chiragjn) [inline ner_v1/api.py:304]: This code block is repeated a lot of times, why not make it a function ?

File: ner_v2/api.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def parse_post_request(request):
         PARAMETER_TIMEZONE: request_data.get('timezone'),
         PARAMETER_LANGUAGE_SCRIPT: request_data.get('language_script', ENGLISH_LANG),
         PARAMETER_SOURCE_LANGUAGE: request_data.get('source_language', ENGLISH_LANG),
+        PARAMETER_PAST_DATE_REFERENCED: request_data.get('date_past_reference', 'False'),
```
Comment (chiragjn) [inline ner_v2/api.py:75]: `False` in string? is this correct?

File: ner_v1/api.py [hunk 3]
```diff
@@ -108,6 +108,31 @@ def parse_post_request(request):
         PARAMETER_PRIOR_RESULTS: request_data.get("predetected_values", [])
     }
 
+    ner_logger.info("parameters dict - {}".format(parameters_dict))
+
+    return parameters_dict
+
+
+def parse_parameters_from_request(request):
+    # type: (django.http.HttpRequest) -> Dict[str, Any]
```
Comment (haptik-deployment) [inline ner_v1/api.py:117]: F821 undefined name 'django'
F821 undefined name 'Dict'
F821 undefined name 'Any'

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

File: ner_v1/api.py (hunk 1 above)
Comment (ankur09011) [reply ner_v1/api.py]: Agree. Pushed the changes

File: ner_v2/api.py (hunk 2 above)
Comment (ankur09011) [reply ner_v2/api.py:75]: Yes, this is correct. We are handling it below.

**Review by chiragjn — APPROVED** [review]

File: ner_v1/api.py [hunk 4]
```diff
@@ -266,8 +285,7 @@ def location(request):
 
     """
     try:
-        parameters_dict = get_parameters_dictionary(request)
-        ner_logger.debug('Start: %s ' % parameters_dict[PARAMETER_ENTITY_NAME])
+        parameters_dict = parameters_dict = parse_parameters_from_request(request)
```
Comment (chiragjn) [inline ner_v1/api.py:288]: Typo here, double assignment

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Review by chiragjn — APPROVED** [review]

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Review by chiragjn — APPROVED** [review]

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Review by amansrivastava17 — APPROVED** [review]

**Pull request merged**